### PR TITLE
[MediaPlayer] add class to set/get adaptive variants and property for buffering time

### PR DESCRIPTION
**Pull request by kqjy777**
[MediaPlayer] add class to set/get adaptive variants and property for buffering time

1.add APIs for setting buffering time
2.add AdaptiveVariants class and APIs to set values

API Changes

If you have the ACR for changing APIs, put the link of the ACR:
•ACR: http://suprem.sec.samsung.net/jira/browse/TCSACR-151

If you don't have the ACR, List all API changes here (or just put None), example:

Added:
 public StreamingBufferingTime BuffTime
 public VariantInfo(int bandwidth, int width, int height)
 public void SetMaxLimit(int bandwidth, int width, int height)
 public VariantInfo GetMaxLimit()
 public IEnumerable AvailableAdaptiveVariants

### CODE REVIEW

File: src/Tizen.Multimedia.MediaPlayer/Player/Player.Properties.cs [hunk 1]
```diff
@@ -24,6 +24,44 @@
...
+        {
+            PreBuffMs = preBuffMs;
+            ReBuffMs = reBuffMs;
+        }
+
+        /// <summary>
+        /// Gets or sets the buffer time to start playback.
+        /// </summary>
+        /// <since_tizen> 5 </since_tizen>
+        public int PreBuffMs
```
Comment (hsgwon) [inline src/Tizen.Multimedia.MediaPlayer/Player/Player.Properties.cs]: How about consider below guideline? 
https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/general-naming-conventions#using-abbreviations-and-acronyms
And make it more clearier.

File: src/Tizen.Multimedia.MediaPlayer/Player/Player.Properties.cs [hunk 2]
```diff
@@ -24,6 +24,44 @@
...
+        {
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// Gets or sets the buffer time during playback if player enter pause state for buffering.
+        /// </summary>
+        /// <since_tizen> 5 </since_tizen>
+        public int ReBuffMs
```
Comment (hsgwon) [inline src/Tizen.Multimedia.MediaPlayer/Player/Player.Properties.cs]: Same above.

File: src/Tizen.Multimedia.MediaPlayer/Player/Player.Properties.cs [hunk 3]
```diff
@@ -105,6 +143,45 @@ public string UserAgent
...
+        /// <exception cref="ObjectDisposedException">The player has already been disposed of.</exception>
+        /// <exception cref="ArgumentOutOfRangeException">
+        ///     <pramref name="PreBuffMs"/> is less than 0.<br/>
+        ///     -or-<br/>
+        ///     <pramref name="ReBuffMs"/> is less than 0.<br/>
+        /// </exception>
+        /// <exception cref="ArgumentException">The value is not valid.</exception>
+        /// <seealso cref="StreamingBufferingTime"/>
+        /// <since_tizen> 5 </since_tizen>
+        public StreamingBufferingTime BuffTime
```
Comment (hsgwon) [inline src/Tizen.Multimedia.MediaPlayer/Player/Player.Properties.cs]: Same above.

File: src/Tizen.Multimedia.MediaPlayer/Player/AdaptiveVariants.cs [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+    public struct VariantInfo
+    {
+        /// <summary>
+        /// Initializes a new instance of the VariantInfo struct.
+        /// </summary>
+        /// <param name="bandwidth">The bandwidth of the stream can be supportable, it must be set. (deafult: -1)</param>
+        /// <param name="width">The width of the stream, this is optional parameter. (deafult: -1)</param>
+        /// <param name="height">The height of the stream, this is optional parameter. (deafult: -1)</param>
+        /// <since_tizen> 5 </since_tizen>
+        public VariantInfo(int bandwidth, int width, int height)
```
Comment (hsgwon) [inline src/Tizen.Multimedia.MediaPlayer/Player/AdaptiveVariants.cs]: If width and height is optional value, how about use defalut parameter?

Comment (kqjy777) [reply src/Tizen.Multimedia.MediaPlayer/Player/AdaptiveVariants.cs]: that's a good idea. i'll fix it.

File: src/Tizen.Multimedia.MediaPlayer/Player/Player.Properties.cs (hunk 1 above)
Comment (kqjy777) [reply src/Tizen.Multimedia.MediaPlayer/Player/Player.Properties.cs]: I've changed it to PreBufferMs and enhanced the description.

File: src/Tizen.Multimedia.MediaPlayer/Player/Player.Properties.cs (hunk 2 above)
Comment (kqjy777) [reply src/Tizen.Multimedia.MediaPlayer/Player/Player.Properties.cs]: I've changed it to ReBufferMs and enhanced the description.

File: src/Tizen.Multimedia.MediaPlayer/Player/Player.Properties.cs (hunk 3 above)
Comment (kqjy777) [reply src/Tizen.Multimedia.MediaPlayer/Player/Player.Properties.cs]: I've changed it to PlayerBufferingTime.

**Pull request merged**